presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraysOverlapFunction.java

```diff
@@ -13,202 +13,300 @@
  */
 package com.facebook.presto.operator.scalar;
 
-import com.facebook.presto.common.NotSupportedException;
 import com.facebook.presto.common.block.Block;
-import com.facebook.presto.common.type.AbstractType;
 import com.facebook.presto.common.type.StandardTypes;
 import com.facebook.presto.common.type.Type;
-import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.operator.aggregation.TypedSet;
 import com.facebook.presto.spi.function.Description;
 import com.facebook.presto.spi.function.OperatorDependency;
 import com.facebook.presto.spi.function.ScalarFunction;
 import com.facebook.presto.spi.function.SqlNullable;
 import com.facebook.presto.spi.function.SqlType;
 import com.facebook.presto.spi.function.TypeParameter;
-import it.unimi.dsi.fastutil.ints.AbstractIntComparator;
-import it.unimi.dsi.fastutil.ints.IntArrays;
-import it.unimi.dsi.fastutil.ints.IntComparator;
+import com.facebook.presto.spi.function.TypeParameterSpecialization;
+import io.airlift.slice.Slice;
 
 import java.lang.invoke.MethodHandle;
-import java.util.Arrays;
 
-import static com.facebook.presto.common.function.OperatorType.LESS_THAN;
-import static com.facebook.presto.common.type.BigintType.BIGINT;
-import static com.facebook.presto.common.type.IntegerType.INTEGER;
-import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.facebook.presto.common.function.OperatorType.EQUAL;
+import static com.facebook.presto.util.Failures.internalError;
 
 @ScalarFunction("arrays_overlap")
 @Description("Returns true if arrays have common elements")
 public final class ArraysOverlapFunction
 {
-    private int[] leftPositions;
-    private int[] rightPositions;
+    private static final int NESTED_LOOP_THRESHOLD = 1000;
```

Comment:
How is this value chosen?

Reply:
I was doing experiments with 100, 500, 1000, 1500, 2000 (coarse grained level) and compared the performance of nested loop and typed set. Then. chose the value where I could see the performance of Nested loop and typed set are almost similar (i.e the cross-over point).

Reply:
It would be nice to put the benchmark data of that in the PR as well. Thanks!